core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -10,33 +10,52 @@
 /**
  * A playing card used in a game of Blackjack.
  */
-final class Card {
-  
-  /**
-   * The rank of a {@link Card}.
-   */
-  enum Rank {}
+public final class Card {
 
-  /**
-   * The suit of a {@link Card}.
-   */
-  enum Suit {}
+  enum Rank {ONE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING, ACE};
+  enum Suit {SPADES, HEARTS, CLUBS, DIAMONDS};
   
-  /**
-   * Gets the rank of this card.
-   * 
-   * @return the rank of this card; never {@literal null}
-   */
-  Rank rank() {
-    return null;
+  // smcdonald4812 - value of the rank, secondary value for the ace, Rank, and Suit
+  private Rank rank;
+  private Suit suit;
+  //smcdonald4812 - could add a String ID here as well with UUID if wanted.
+  
+  //smcdonald4812 - default constructor, non-null assignment.
+  
+  public Card() {
+    super();
+    this.rank = Rank.ONE;
+    this.suit = Suit.SPADES;
+  }
+  
+  //smcdonald4812 - Rank constructor must call this() as one has been provided.  .ordinal() is zerobased enum indexing.
+  
+  public Card(Rank rank) {
+    this();
+    this.rank = rank;
+  }
+  
+  /*
+  * smcdonald4812 - Suit, Rank constructor.
+  */
+  
+  public Card(Rank rank, Suit suit) {
+    this(rank);
+    this.suit = suit;
   }
   
   /**
-   * Gets the suit of this card.
+   * Gets the rank of this card.
    * 
-   * @return the suit of this card; never {@literal null}
+   * @return the rank of this card; never {@literal null}
+   * smcdonald4812 - non-null in this(). There are no setters as Card shouldn't be changed once it is created.
    */
-  Suit suit() {
-    return null;
+   
+   public Rank getRank() {
```

Comment:
Any reason for this rename? It's not a problem, I'm just curious.

Reply:
I've been studying for the OCA, and it stats that all getters and setters must be either get, set, or is. I just thought that was standard practice?

Reply:
That makes sense. I just thought it was one of those things like camelCase.